### PR TITLE
apiserver: refactor handleError in endpoints/filters

### PR DESCRIPTION
**Pull request by tkashem**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

In support of https://github.com/kubernetes/kubernetes/pull/114189

It's a small refactor of the `handleError` function inside `endpoints/filters` package

#### Which issue(s) this PR fixes:
Fixes #

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by tkashem** [comment]
/cc @wojtek-t 

(for https://github.com/kubernetes/kubernetes/pull/114189)

**Comment by tkashem** [comment]
/assign @sttts 

PTAL, it's a small refactor PR in support of https://github.com/kubernetes/kubernetes/pull/114189

File: staging/src/k8s.io/apiserver/pkg/endpoints/filters/request_deadline.go [hunk 1]
```diff
@@ -166,8 +166,12 @@ func parseTimeout(req *http.Request) (time.Duration, bool, error) {
...
-	http.Error(w, errorMsg, code)
-	klog.Errorf(errorMsg)
+// handleError does the following:
+// a) it writes the specified error code, and msg to the ResponseWriter
+// object, it does not print the given err into the ResponseWriter object.
+// b) additionally, it prints the given msg, and err to the log with other
+// request scoped data that helps identify the given request.
+func handleError(w http.ResponseWriter, r *http.Request, code int, msg string, err error) {
+	http.Error(w, msg, code)
+	klog.ErrorS(err, msg, "method", r.Method, "URI", r.RequestURI, "auditID", audit.GetAuditIDTruncated(r.Context()))
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/endpoints/filters/request_deadline.go:176]: This seems wrong in the current way you're passing err = nil everywhere.

We should use a non-nil error here...

**Pull request merged**